### PR TITLE
Renamed a TensorListMetaData property. Cleaned up a test

### PR DESCRIPTION
**Pull request by izdeby**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46664 Move device guard from MultiTensorApply.cuh
* #46663 Remove .pyi from /_multi_tensor
* **#46662 Renamed a TensorListMetaData property. Cleaned up a test**
* #46661 Refactor arguments checks for foraech APIs

Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

### CODE REVIEW

File: test/test_foreach.py [hunk 1]
```diff
@@ -703,56 +703,57 @@ def test_bin_op_scalar_with_different_tensor_dtypes(self, device):
...
+        for bin_op, bin_op_ in zip(self.foreach_bin_ops, self.foreach_bin_ops_):
+            tensors1 = []
+            tensors2 = []
+
+            # Empty lists
+            with self.assertRaises(RuntimeError):
+                bin_op(tensors1, tensors2)
+            with self.assertRaises(RuntimeError):
+                bin_op_(tensors1, tensors2)
+
```
Comment (zou3519) [inline test/test_foreach.py:716]: nit: prefer AssertRaisesRegex so that we can check that the error message is sane

File: test/test_foreach.py [hunk 2]
```diff
@@ -703,56 +703,57 @@ def test_bin_op_scalar_with_different_tensor_dtypes(self, device):
...
+            # different devices
+            if torch.cuda.is_available() and torch.cuda.device_count() > 1:
+                tensor1 = torch.zeros(10, 10, device="cuda:0")
+                tensor2 = torch.ones(10, 10, device="cuda:1")
+                with self.assertRaisesRegex(RuntimeError, "Expected all tensors to be on the same device"):
+                    bin_op([tensor1], [tensor2])
+                with self.assertRaisesRegex(RuntimeError, "Expected all tensors to be on the same device"):
+                    bin_op_([tensor1], [tensor2])
+
+            # Coresponding tensors with different sizes
```
Comment (zou3519) [inline test/test_foreach.py:750]: nit: "coresponding" -> "corresponding"

**Review by zou3519 — APPROVED** [review]

**Comment by zou3519** [comment]
btw, tests are failing with:
```
Oct 23 20:41:06 ====================================================================== 
Oct 23 20:41:06 FAIL [0.016s]: test_bin_op_list_error_cases_cpu (__main__.TestForeachCPU) 
Oct 23 20:41:06 ---------------------------------------------------------------------- 
Oct 23 20:41:06 RuntimeError: Scalars list must have at least one value. 
Oct 23 20:41:06  
Oct 23 20:41:06 During handling of the above exception, another exception occurred: 
Oct 23 20:41:06  
Oct 23 20:41:06 Traceback (most recent call last): 
Oct 23 20:41:06   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 274, in instantiated_test 
Oct 23 20:41:06     result = test_fn(self, *args) 
Oct 23 20:41:06   File "test_foreach.py", line 720, in test_bin_op_list_error_cases 
Oct 23 20:41:06     bin_op(tensors1, tensors2) 
```

File: test/test_foreach.py (hunk 2 above)
Comment (izdeby) [reply test/test_foreach.py:750]: Done

File: test/test_foreach.py [hunk 3]
```diff
@@ -799,33 +800,6 @@ def test_div_list(self, device, dtype):
...
-            with self.assertRaises(RuntimeError):
-                bin_op(tensors1, tensors2)
-
-            # One empty list
-            tensors1.append(torch.tensor([1], device=device))
-            with self.assertRaises(RuntimeError):
-                bin_op(tensors1, tensors2)
-
-            # Lists have different amount of tensors
-            tensors2.append(torch.tensor([1], device=device))
```
Comment (ngimel) [inline test/test_foreach.py:817]: I don't see this test anymore - did I miss it?

**Review by ngimel — APPROVED** [review]

File: test/test_foreach.py (hunk 3 above)
Comment (izdeby) [reply test/test_foreach.py:817]: These test cases were duplicated. You can see them in `test_bin_op_list_error_cases`

Comment (ngimel) [reply test/test_foreach.py:817]: Oh thanks, I'm blind.

**Pull request closed without merging**